src/PactNet/Verifier/IPactVerifier.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace PactNet.Verifier
+{
+    /// <summary>
+    /// Pact verifier
+    /// </summary>
+    public interface IPactVerifier
+    {
+        /// <summary>
+        /// Set the provider details
+        /// </summary>
+        /// <param name="providerName">Name of the provider</param>
+        /// <param name="pactUri">URI of the running service</param>
+        /// <returns>Fluent builder</returns>
+        IPactVerifierProvider ServiceProvider(string providerName, Uri pactUri);
```

Comment:
Hm, okay, I see where you are going here. Therefore, you could add a IMessagePactVerifierProvider that would eventually return another state builder.